### ISSUE TITLE
Case-insensitive procedures

### ISSUE THREAD

**Issue by lassik** [pre-srfi/substring#2]
We should probably have `-ci` variants of the procedures.

**Comment by johnwcowan** [comment]
That complicates things, though, because now there are matches like ß = ss, which means that the matches are not all identical, so we have to deal in start-end pairs rather than just starts. I'm not convinced this is necessary.

**Comment by lassik** [comment]
You're right. It would mess with the entire index-based API. At a minimum, we'd have to use (start . end) index pairs instead of only the start index.

**Comment by lassik** [comment]
Would it be blasphemous to have `-ascii-ci` procedures?
